### ISSUE TITLE
Create Appointment with wrong client!

### ISSUE THREAD

**Issue by klondikemarlen** [elthran/BookingSystem#19]
While creating a new appointment using the "client book an appointment link". I created an appointment with the same email address but a different name. It created an appointment for the first owner of the email.

 This should have thrown an Error saying there was a Name/email mismatch ... or that that email was already in use.

This would let someone spam your appointment list.

**Comment by elthran** [comment]
Good point! The nuances of this page are bothering me. If you create a new appointment but type your name differently (ie:)
Jacob
jacobbrunner@gmail.com

and then
Jacob Brunner
jacobbrunner@gmail.com

It should both be under the same client and not throw any errors (because Melissa says 100% priority is ease of use and if customers have any technical troubles at all they often just stop trying to book the appointment and give up). Now solutions are hard. I'm not sure of a good way to solve this. However, there are some positives:
- email notification is sent and the email holder needs to confirm the booking
- we could put a timer on how frequently you can book. therefore after several bookings it times out for like 15 minutes. should stop any abuse?

Does that sound like a working solution?

**Comment by klondikemarlen** [comment]
That sounds reasonable. I'll see what I can do.

**Comment by klondikemarlen** [comment]
I'm pretty sure I could modify my search function (latest Pull request https://github.com/elthran/BookingSystem/pull/25) to return a "best match" for this client ... then ask for confirmation?
Q: Is this you?
A: nope ... try again
A: yes ... overwriting ...
